pip/commands/list.py

```diff
@@ -146,15 +153,20 @@ def run(self, options, args):
             raise CommandError(
                 "Options --outdated and --uptodate cannot be combined.")
 
+        # TODO (nvdv): Check options combinations.
+
```

Comment:
Now `self.get_no_deps_only` acts as final filter for other options. Let me know If I missed somethin.